### DISCUSSION TITLE
Neural Network models do not complete in the Console and jupyter notebook

### DISCUSSION THREAD

**Discussion by Santoshjnawali** [posit-dev/positron#7896] [Q&A]
I am noticing neural network models get stuck in the console or notebook. However, it properly runs in the terminal. I am using TensorFlow. Not sure if this is my device-specific issue.

**Comment by juliasilge** [comment]
Can you create a a minimal reproducible example of the code you are trying to run? The goal of this kind of example is to make it easier for us to recreate your problem so that we can understand it and/or fix it. If you are using R to run TensorFlow, you may want to start with the [reprex](https://reprex.tidyverse.org/) package as a way to create this kind of small, self-contained example. Thanks! 🙌

**Reply by Santoshjnawali** [reply]
Here is an example. 

import tensorflow as tf
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Flatten, Dense
from tensorflow.keras.utils import to_categorical

# Load and preprocess the data
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0  # Normalize
y_train = to_categorical(y_train)
y_test = to_categorical(y_test)

# Create a simple neural network model
model = Sequential([
    Flatten(input_shape=(28, 28)),  # Convert 2D to 1D
    Dense(128, activation='relu'),
    Dense(10, activation='softmax')  # 10 output classes
])

# Compile the model
model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# Train the model for 10 epochs
model.fit(x_train, y_train, epochs=10, batch_size=32, validation_data=(x_test, y_test))


If I run this code in the console, it gets stuck in the console at epoch 1, and the same with Jupyter Notebook inside Positron. However, it works fine if I run the file in the terminal.

**Comment by juliasilge** [comment]
Hmmmm, I'm not able to reproduce this problem:

![Screenshot 2025-06-03 at 12 47 51 PM](https://github.com/user-attachments/assets/35f21939-9ff7-467c-ae33-5da4740d539e)

Can you share more details about the version of Positron you are using (copy from the `Positron > About` menu) and your platform?

**Reply by Santoshjnawali** [reply]
I'm on macOS

Positron Version: 2025.07.0 (Universal) build 49
Code - OSS Version: 1.100.0
Commit: 8b3e95ab7926c0643351147fdbcc1145ab8f5510
Date: 2025-06-05T03:33:36.609Z
Electron: 34.5.1
Chromium: 132.0.6834.210
Node.js: 20.19.0
V8: 13.2.152.41-electron.0
OS: Darwin arm64 24.5.0

**Reply by juliasilge** [reply]
Hmmmm, I would expect that to work just fine. 🤔 

- Can you check out [the Python console and maybe also kernel logs](https://positron.posit.co/troubleshooting.html#python-and-r-logs) from the output channels, and see what might be happening there?
- Can you share a screenshot or screen recording so we can see more details of what is happening?

**Reply by Santoshjnawali** [reply]
<img width="1436" alt="Screenshot 2025-06-06 at 3 15 02 PM" src="https://github.com/user-attachments/assets/74a3ca7c-e769-4f6b-8ad3-f17c13a570b6" />

This is what I see in the output channel and in the console. It does not progress after showing epoch 1/10.

**Reply by juliasilge** [reply]
Unfortunately we are still not able to reproduce this problem. Can you update your `python.languageServerLogLevel` setting to "debug", restart Positron, and then share your Python _kernel_ logs (not console) when you try to run this code and it gets stuck?

**Reply by Santoshjnawali** [reply]
It seems to be working now with the latest updates
